common/src/main/java/tds/common/cache/NameAwareGuavaCacheManager.java

```diff
@@ -0,0 +1,47 @@
+package tds.common.cache;
+
+import com.google.common.cache.CacheBuilder;
+import org.springframework.cache.Cache;
+import org.springframework.cache.guava.GuavaCache;
+import org.springframework.cache.guava.GuavaCacheManager;
+
+import java.util.Arrays;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+/**
+ * This implementation of a GuavaCacheManager configures internal caches based upon the cache name.
+ */
+public class NameAwareGuavaCacheManager extends GuavaCacheManager {
+
+    private final CacheConfigurationProperties cacheProperties;
+
+    /**
+     * Constructor
+     *
+     * @param cacheProperties   Configuration cache properties
+     * @param cacheNames        The cache names
+     */
+    public NameAwareGuavaCacheManager(final CacheConfigurationProperties cacheProperties,
+                                      final String... cacheNames) {
+        this.cacheProperties = cacheProperties;
+        this.setCacheNames(Arrays.asList(cacheNames));
+    }
+
+    @Override
+    protected Cache createGuavaCache(final String name) {
```

Comment:
This method is called by the GuavaCacheManager when trying to find a non-existent cache by name.  The default implementation creates a basic Guava cache.  This override configures the created cache based upon the name.